### DISCUSSION TITLE
Scanorama input order

### DISCUSSION THREAD

**Discussion by rvernica** [brianhie/scanorama#129] [Q&A]
We ran Scanorama with three matrices. We noticed different run-times depending on the input order. Is this expected? Should we expect significant differences between the resulting outputs?

Aditionally, one of the matrices is rather _skinny_ with only 21 genes. Does this type of input make sense for the Scanorama algorithm?

Here are the three matrices, the order in which we passed them to Scanorama and the corresponding run-times:

1. `<7087x23730>`, `<12971x25241>`, `<286256x21>` 2294.25 seconds
1. `<286256x21>`, `<7087x23730>`, `<12971x25241>` 3136.87 seconds
1. `<286256x21>`, `<12971x25241>`, `<7087x23730>` 2101.02 seconds

Notice a 50% run-time difference between run (2) and (3).

**Comment by brianhie** [comment]
I'm not sure, I would not expect enormous differences. Have you seen if these timings hold across multiple runs?

And I would expect Scanorama to work on 21 genes, though one thing to keep in mind is that the code integrates the intersection of all sets of genes, so any higher-dimensional matrices would be reduced to 21 or fewer dimensions.

**Reply by rvernica** [reply]
Thank you. Yes, we did multiple runs and the timing held.

**Comment by rvernica** [comment]
We looked at the generated output figures given different input orders. We used the [example](https://github.com/brianhie/scanorama#examples-from-paper) from the paper as described in the repository README. We altered the input order by altering the oeder of the elements in this list:
```
data_names = [
    'data/293t_jurkat/jurkat',
    'data/293t_jurkat/293t',
    'data/293t_jurkat/jurkat_293t_50_50',
]
```
We tried four different orders and generated the corresponding _ds_ visualizations. They are:
1. _jurkat, 293t, jurkat_293t_50_50_
   ![293t_jurkat_ds_j-2-j2](https://user-images.githubusercontent.com/241185/198755871-74a0053f-b77f-4272-ac87-f58ae3d33e51.png)
1. _jurkat_293t_50_50, 293t, jurkat_
   ![293t_jurkat_ds_j2-2-j](https://user-images.githubusercontent.com/241185/198755897-e9af470b-5702-402d-acb0-aee20caab492.png)
1. _293t, jurkat, jurkat_293t_50_50_
   ![293t_jurkat_ds_2-j-j2](https://user-images.githubusercontent.com/241185/198755911-06e9cb4e-9e73-477e-957e-519a280c1f14.png)
1. _293t, jurkat_293t_50_50, jurkat_
   ![293t_jurkat_ds_2-j2-j](https://user-images.githubusercontent.com/241185/198755922-bbdc830a-5ec9-4d74-b9b1-a82d5aa383f4.png)

I guess it is hard to say if changing the order changes the output.
